NitroxServer/Serialization/ServerProtobufSerializer.cs

```diff
@@ -41,14 +43,15 @@ public void Deserialize(Stream stream, object o, Type t)
 
         private void RegisterHardCodedTypes()
         {
-            Model.Add(typeof(UnityEngine.Light), true);
-            Model.Add(typeof(UnityEngine.BoxCollider), true);
-            Model.Add(typeof(UnityEngine.SphereCollider), true);
-            Model.Add(typeof(UnityEngine.MeshCollider), true);
-            Model.Add(typeof(UnityEngine.Vector3), false).SetSurrogate(typeof(NitroxVector3));
-            Model.Add(typeof(UnityEngine.Quaternion), false).SetSurrogate(typeof(NitroxQuaternion));
-            Model.Add(typeof(UnityEngine.Transform), false).SetSurrogate(typeof(NitroxTransform));
-            Model.Add(typeof(UnityEngine.GameObject), false).SetSurrogate(typeof(UnityStubs.GameObject));            
+            Assembly unity = AppDomain.CurrentDomain.GetAssemblies().First(assembly => assembly.GetName().Name == "UnityEngine");
```

Comment:
What is the purpose of doing this at runtime vs at compile time?  Also, aren't there now a lot unity assemblies?  Are all of these located in the same one?

Reply:
You're correct. Didn't know how this worked previously.